### PR TITLE
Update guestbook example to remove inappropriate terminology

### PR DESCRIPTION
**Pull request by jimangel**
Opening this PR to complete efforts that were started in https://github.com/kubernetes/examples/pull/393

I need to do a tech review and can address any feedback. If WG Naming agrees, https://github.com/kubernetes/examples/pull/393 can be closed in favor of this PR.

Many thanks to @paulczar for doing the heavy lifting!

/cc @kubernetes/wg-naming 

>The guestbook and redis documentation are littered with master/slave terminology. This update along with the update to the [website](https://github.com/kubernetes/website/pull/22934) removes that terminology and technology.
>
>Addresses https://github.com/kubernetes/website/issues/22918
>
>Signed-off-by: Paul Czarkowski

### CODE REVIEW

File: guestbook/php-mongodb/guestbook.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
+<?php
+
+error_reporting(E_ALL);
```
Comment (kbhawkey) [inline guestbook/php-mongodb/guestbook.php:3]: Is this file content ported from `guestbook/php-redis/guestbook.php`?

File: guestbook/frontend-deployment.yaml [hunk 2]
```diff
@@ -15,8 +18,10 @@ spec:
         tier: frontend
     spec:
       containers:
-      - name: php-redis
-        image: gcr.io/google-samples/gb-frontend:v4
+      - name: guestbook
+        image: paulczar/gb-frontend:v5
```
Comment (kbhawkey) [inline guestbook/frontend-deployment.yaml:22]: Is this a private image? How will the image be maintained?

Comment (celestehorgan) [reply guestbook/frontend-deployment.yaml:22]: The plan is to make it a non-private image – I assume there's still wok to be done, @jimangel?

**Comment by kbhawkey** [comment]
👋 @jimangel . Do you need help to move this forward?

**Comment by jimangel** [comment]
> 👋 @jimangel . Do you need help to move this forward?

This completely fell off my plate. I will make progress this weekend. In the meantime, I would love any help from folks who have time to test functionality.

**Comment by celestehorgan** [comment]
@jimangel – I'll take a look at this later today!

**Comment by jimangel** [comment]
Sorry @celestehorgan it took awhile to sit down and test but I finally completed it. There's a few things we'll need to update in https://kubernetes.io/docs/tutorials/stateless-application/guestbook/ but overall:

- **BUG (update `guestbook.md` in k/website)**
  - `app.kubernetes.io/name=guestbook` becomes `app=guestbook`
  - `app.kubernetes.io/name=mongo` becomes `app=mongo`

- **BUG (no action)**
  - The guestbook app started duplicating entries when I scaled to 5 replicas. It eventually stopped, but I'm not sure what was going on. It's probably something in the frontend code. For a `hello-world` app it probably doesn't matter...

Following the readme, I was able to get the service up and running:
image

I will poke around to see how we can get the image updated. For now, the image `paulczar/gb-frontend:v5` is public but it would be better to update `gcr.io/google-samples/gb-frontend:v4` to `gcr.io/google-samples/gb-frontend:v5`.

Alternately, we could include building the image as part of the tutorial (build and push to public repo?).

I would be happy with this merging as-is and fixes could follow. Thanks!

File: guestbook/frontend-deployment.yaml (hunk 2 above)
Comment (jimangel) [reply guestbook/frontend-deployment.yaml:22]: Addressed in https://github.com/kubernetes/examples/pull/410#issuecomment-861155813 and open to feedback!

File: guestbook/php-mongodb/guestbook.php (hunk 1 above)
Comment (jimangel) [reply guestbook/php-mongodb/guestbook.php:3]: It [does appear so](https://github.com/kubernetes/examples/blob/master/guestbook/php-redis/guestbook.php), but I'm not sure. Why do you ask?

**Comment by jimangel** [comment]
/hold

Things got a little weird... It turns out the samples originated from [GCP docs](https://cloud.google.com/kubernetes-engine/docs/tutorials/guestbook) and already have addressed the inappropriate terminology in [a PR](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/pull/127) a year ago. Also, The frontend image is already updated to v5 and published. For curious minds, the image's source of truth is in this repo's [cloudbuild](https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/blob/master/guestbook/php-redis/cloudbuild.yaml). The modifications continue use of redis and have changed `slave/master` to `follow/leader`.

The foundation of this PR is already merged in k/website including the move to mongo.

Considering the docs / image are being actively maintained, we shouldn't increase tech debt. As a result, I plan on doing the following:
- [ ] Change this PR to match the GCP yaml (and clean up as applicable)
- [ ] Add comments to impacted k/website and k/example files referencing the "source of truth"
- [ ] Create a PR to address the above in k/website too (yay triple sourced content 😢 )

/cc @kubernetes/sig-docs-en-owners

**Pull request closed without merging**